src/test/java/org/jabref/logic/integrity/DoiDuplicationCheckerTest.java

```diff
@@ -0,0 +1,66 @@
+package org.jabref.logic.integrity;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DoiDuplicationCheckerTest {
+
+    private final DoiDuplicationChecker checker = new DoiDuplicationChecker();
+    private BibEntry entry1;
+    private BibEntry entry2;
+    private BibEntry entry3;
+    private BibEntry entry4;
+    private BibEntry entry5;
+
+    @BeforeEach
+    void setUp() {
+        // entry1 and entry2 have duplicate DOI
+        entry1 = new BibEntry().withField(StandardField.DOI, "10.1023/A:1022883727209");
+        entry2 = new BibEntry().withField(StandardField.DOI, "10.1023/A:1022883727209");
+
+        // entry3 and entry4 have duplicate DOI
+        entry3 = new BibEntry().withField(StandardField.DOI, "10.1177/1461444811422887");
+        entry4 = new BibEntry().withField(StandardField.DOI, "10.1177/1461444811422887");
+
+        // entry5 has a different DOI than the above
+        entry5 = new BibEntry().withField(StandardField.DOI, "10.1145/2568225.2568315");
+
```

Comment:
Move that to the field declaration.

Try to use more readable names:

- `entry1` -> `doiA_entry1`.
- `entry2` -> `doiA_entry2`.
- `entry1` -> `doiB_entry1`.
- `entry2` -> `doiB_entry2`.

Reply:
Code is updated as suggested. :)